src/librustc_mir/borrow_check/error_reporting.rs

```diff
@@ -519,14 +539,33 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
         proper_span: Span,
         _end_span: Option<Span>
     ) {
+        debug!(
+            "report_unscoped_temporary_value_does_not_live_long_enough(\
+                {:?}, {:?}, {:?}, {:?}, {:?}\
+            )",
+            context,
+            scope_tree,
+            borrow,
+            drop_span,
+            proper_span
+        );
+
         let mut err = self.tcx.path_does_not_live_long_enough(proper_span,
                                                               "borrowed value",
                                                               Origin::Mir);
         err.span_label(proper_span, "temporary value does not live long enough");
         err.span_label(drop_span, "temporary value only lives until here");
-        self.tcx.note_and_explain_region(scope_tree, &mut err,
-                                         "borrowed value must be valid for ",
-                                         borrow.region, "...");
+
+        if !self.tcx.sess.nll() {
```

Comment:
@nikomatsakis this is what I was talking about, unsure what kind of example triggers a call to `report_unscoped_temporary_value_does_not_live_long_enough`. Need to investigate :slightly_smiling_face: